cpp/src/arrow/filesystem/gcsfs.cc

```diff
@@ -332,12 +385,15 @@ class GcsFileSystem::Impl {
   static Result<FileInfo> GetFileInfoImpl(const GcsPath& path,
                                           const google::cloud::Status& status,
                                           FileType type) {
+    const auto& canonical = type == FileType::Directory
+                                ? internal::EnsureTrailingSlash(path.full_path)
+                                : path.full_path;
```

Comment:
Sorry, I never remember for sure: is it actually ok to keep a const-ref to a temporary?

Reply:
Yes, it is ok:

> The lifetime of a temporary object may be extended by binding to a const lvalue reference or to an rvalue reference (since C++11), see reference initialization for details.

https://en.cppreference.com/w/cpp/language/lifetime#:~:text=The%20lifetime%20of%20a%20temporary%20object%20may%20be%20extended%20by%20binding%20to%20a%20const%20lvalue%20reference

https://en.cppreference.com/w/cpp/language/reference_initialization#Lifetime_of_a_temporary

Reply:
Ha, thanks. I always get confused by this.